File: my-folder/0001-two-sum/solution.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numsDict = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            if (numsDict.ContainsKey(target - nums[i])) {
                return new int[2] {i, numsDict[target - nums[i]]};
            } else {
                if (!numsDict.ContainsKey(nums[i])) {
                    numsDict.Add(nums[i], i);
                }
            }
        }
        return null;
    }
}

File: my-folder/0001-two-sum/solution.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> storage = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];
            if (storage.containsKey(difference)) {
                return new int[] {storage.get(difference), i};
            } else {
                storage.put(nums[i], i);
            }
        }
        return null;
    }
}
